### PR TITLE
Add multiplayer joystick example

### PR DESCRIPTION
**Pull request by KentAugust**
An example using the ```pygame.JOYDEVICEADDED``` and ```pygame.JOYDEVICEREMOVED``` events for hot plugging.

* The player can change its color if still hasn't joined.
* After player has joined, he can move freely.
* When a controller is connected a new player will appear (I set a limit of two player, so a third controller wont affect).
* A message will be printed when connect/disconnect.
* I message will be displayed in the middle of the screen If there isn't  a single controller connected.

**Input**
* *A*: button 0
* *B*: button 1
* *Joystick*: axis 0 and 1
 
**Example video**

https://github.com/pygame-community/pygame-ce/assets/97701247/35da026f-5a57-4ece-bf79-4539d32a1a26

### CODE REVIEW

**Comment by bilhox** [comment]
Hello,

Good work for the example, by the way you should use ``if else`` for events rather than pattern matching for retro compatibility with python <=3.9

**Comment by KentAugust** [comment]
I forgot that match was introduced in 3.10. I changed ^^

File: examples/multiplayer_joystick.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+def create_surf(size, color):
+    surf = pygame.Surface(size)
+    surf.fill(color)
+    return surf
+
+
+pygame.init()
+
+WIDHT, HEIGHT = 500, 500
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:47]:
```suggestion
WIDTH, HEIGHT = 500, 500
```

File: examples/multiplayer_joystick.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+
+def connect_joystick(index):
+    if len(active_players) < len(players):
+        joy = pygame.Joystick(index)
+        index = players.index(None)
+        # player data
+        players[index] = {
+            "joy": joy,
+            "pos": [WIDHT * 0.25 + index * 64, 128],
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:12]:
```suggestion
            "pos": [WIDTH * 0.25 + index * 64, 128],
```

File: examples/multiplayer_joystick.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                if event.button == 0:
+                    index = active_players[event.instance_id]
+                    players[index]["joined"] = True
+                    print(f"P{index + 1} joined")
+                # leave player
+                if event.button == 1:
+                    index = active_players[event.instance_id]
+                    if players[index]["joined"]:
+                        players[index]["joined"] = False
+                        players[index]["pos"] = [WIDHT * 0.25 + index * 64, 128]
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:97]:
```suggestion
                        players[index]["pos"] = [WIDTH * 0.25 + index * 64, 128]
```

File: examples/multiplayer_joystick.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                    if not player["joined"]:
+                        if event.value >= 1.0:
+                            player["surf_idx"] += 1
+                        elif event.value <= -1.0:
+                            player["surf_idx"] -= 1
+                        player["surf_idx"] = player["surf_idx"] % len(colors)
+                        player["surf"] = colors[player["surf_idx"]]
+
+    screen.fill((30, 30, 30))
+    pygame.draw.line(screen, (230, 230, 230), (0, 96), (WIDHT, 96), 2)
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:114]:
```suggestion
    pygame.draw.line(screen, (230, 230, 230), (0, 96), (WIDTH, 96), 2)
```

File: examples/multiplayer_joystick.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    # update and draw players
+    for player in players:
+        if player:
+            controll_player(player)
+            screen.blit(player["surf"], player["pos"])
+
+    # draw avalible colors
+    for i, surf in enumerate(colors):
+        screen.blit(surf, (WIDHT * 0.25 + i * 64, 32))
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:124]:
```suggestion
        screen.blit(surf, (WIDTH * 0.25 + i * 64, 32))
```

File: examples/multiplayer_joystick.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+    # show message for connecting a controller
+    if zero_joy_connected:
+        screen.blit(
+            font_b.render(
+                "Please connect a controller.", True, (230, 230, 230), None, 500 - 20
+            ),
+            (WIDHT * 0.3, HEIGHT * 0.5),
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:132]:
```suggestion
            (WIDTH * 0.3, HEIGHT * 0.5),
```

File: examples/multiplayer_joystick.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        )
+
+    screen.blit(
+        font_a.render(
+            "A: join    B: leave    Joystick: move / change color",
+            True,
+            (230, 230, 230),
+            None,
+            WIDHT - 20,
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:141]:
```suggestion
            WIDTH - 20,
```

**Review by MyreMylar — COMMENTED** [review]
LGTM 👍 Nice demo video too.

I noticed you misspelled `WIDTH`, so I've just highlighted that.

**Review by MyreMylar — APPROVED** [review]

File: examples/multiplayer_joystick.py (hunk 7 above)
Comment (KentAugust) [reply examples/multiplayer_joystick.py]: Thanks, sometimes that happen when write fast

File: examples/multiplayer_joystick.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+
+def disconnect_joystick(instance_id: int):
+    index = active_players[instance_id]
+    players[index] = None
+    del active_players[instance_id]
+    print(f"P{index + 1} Disconnected")
+
+
+def controll_player(player):  # move player
```
Comment (novialriptide) [inline examples/multiplayer_joystick.py:28]: typo `controll`

File: examples/multiplayer_joystick.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    screen.fill((30, 30, 30))
+    pygame.draw.line(screen, (230, 230, 230), (0, 96), (WIDTH, 96), 2)
+
+    # update and draw players
+    for player in players:
+        if player:
+            controll_player(player)
+            screen.blit(player["surf"], player["pos"])
+
+    # draw avalible colors
```
Comment (novialriptide) [inline examples/multiplayer_joystick.py:119]: typo: `avalible`

File: examples/multiplayer_joystick.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    for i, surf in enumerate(colors):
+        screen.blit(surf, (WIDTH * 0.25 + i * 64, 32))
+
+    # show message for connecting a controller
+    if zero_joy_connected:
```
Comment (novialriptide) [inline examples/multiplayer_joystick.py:124]: variable could be renamed to `are_no_controllers_connected`

Comment (KentAugust) [reply examples/multiplayer_joystick.py]: Thanks, that make more sense

File: examples/multiplayer_joystick.py [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+pygame.init()
+
+WIDTH, HEIGHT = 500, 500
+screen = pygame.display.set_mode((WIDTH, HEIGHT))
+pygame.display.set_caption("Multiplayer Joystick example")
+clock = pygame.Clock()
+font_b = pygame.font.SysFont(None, 25)
+font_a = pygame.font.SysFont(None, 16)
+
+players = [None, None] # two players limit
```
Comment (MyreMylar) [inline examples/multiplayer_joystick.py:51]:
```suggestion
players = [None, None]  # two players limit
```

**Review by Starbuck5 — APPROVED** [review]
Thanks!

**Pull request merged**